pkg/cli/cli.go

```diff
@@ -353,7 +353,7 @@ func validatePlugins(plugins ...plugin.Base) error {
 		}
 		// Check for duplicate plugin names. Names outside of a version can
 		// conflict because multiple project versions of a plugin may exist.
-		if _, seen := pluginNameSet[pluginName]; seen {
+		if _, seen := pluginNameSet[fmt.Sprintf("%v/%v", p.Name(), p.Version())]; seen {
```

Comment:
Fix plugin check for duplicates.

Reply:
NOTE: The changes performed here to introduce the V3 plugin which was discussed were evaluated in https://github.com/kubernetes-sigs/kubebuilder/pull/1530 where the test is done by moving with a new MINOR bump 3.1.0 as follow up of this one which would scaffold a new file .dockerignore.